### DISCUSSION TITLE
Unable to use secrets inside an action of a workflow

### DISCUSSION THREAD

**Discussion by yantolee** [community/community#24946] [Actions]
<p>I am trying to use secrets inside an action yml file but getting error as follow : Unrecognized named-value: ‘secrets’. Located at position 1 within expression: secrets.server_address</p>
<p>But there isn’t such error when I using the secrets directly inside the workflow yml file</p>
<p>Is it that it is not possible to use secrets inside action file? Thanks</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>If you want to use a secret in an action, you have to explicitly pass it as input or environment variable. I assume that’s a security measure to prevent actions from stealthily stealing secrets, at least that’s what I’d worry about otherwise.</p>
<p>See also:</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/actions/reference/encrypted-secrets#accessing-your-secrets" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/actions/reference/encrypted-secrets#accessing-your-secrets" target="_blank" rel="noopener">Encrypted secrets - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/actions/reference/encrypted-secrets#accessing-your-secrets" target="_blank" rel="noopener">//docs.github.com/en/actions/reference/encrypted-secrets</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by yantolee** [comment]
<p>I see noted that it need to be explicitly passed to the action. In that case, it seems like it will defeat the purpose for my use case.</p>
<p>I am trying to convert a step of sending email which will involved making use of secrets (for the credentials of the email account) into a reusable action rather than wring same chunck of code for every of my workflow which require email sending.</p>
<p>Thanks for clarifying in terms of the secrets usage inside an action.</p>
